google/gax/__init__.py

```diff
@@ -482,3 +489,160 @@ def __next__(self):
         if self._index >= len(self._current):
             self._current = None
         return resource
+
+
+def _from_any(pb_type, any_pb):
+    if not any_pb.Is(pb_type.DESCRIPTOR):
+        raise TypeError(
+            'Could not convert {} to {}'.format(
+                any_pb.__class__.__name__, pb_type.__name__))
+
+    return pb_type.FromString(any_pb.value)
```

Comment:
I see. You based it off of `google-cloud-python` code. Let's check this with @dhermes. That code maintains its own registry of proto types, whereas we should be using the gRPC registry at the GAX level. I'm not sure if that dictates any code changes, especially since we know exactly what types the result and metadata are.

And yes, the result and metadata will be protos for the foreseeable future.